etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/database/PostgresDaoTest.groovy

```diff
@@ -137,4 +137,59 @@ class PostgresDaoTest extends Specification {
         then:
         thrown(SQLException)
     }
+
+    def "fetchMetadata returns null when rows  do not exist"() {
+        given:
+        def mockDriver = Mock(SqlDriverManager)
+        def mockPreparedStatement = Mock(PreparedStatement)
+        def mockResultSet = Mock(ResultSet)
+        def expected = null
+        Connection mockConn = Mock(Connection)
+
+        mockDriver.getConnection(_ as String, _ as Properties) >> mockConn
+        mockConn.prepareStatement(_ as String) >>  mockPreparedStatement
+        mockResultSet.next() >> false
+        mockPreparedStatement.executeQuery() >> mockResultSet
+
+        TestApplicationContext.register(SqlDriverManager, mockDriver)
+        TestApplicationContext.injectRegisteredImplementations()
+
+        when:
+        def actual = PostgresDao.getInstance().fetchMetadata("mock_lookup")
+
+        then:
+        actual == expected
+    }
+
+    def "fetchMetadata returns partnermetadata when rows exist"() {
+        given:
+        def mockDriver = Mock(SqlDriverManager)
+        def mockPreparedStatement = Mock(PreparedStatement)
```

Comment:
We can put mocks setup in "def setup() {}" on the top.

Reply:
Yes, we can.

Reply:
Done.